File: articles/behavioral-design-technology-systems/go/interface_welfare.go
package main

import (
	"fmt"
	"math"
)

func userWelfare(joined, baselineValue, rewardIntensity, frictionAsymmetry, autonomyPreference, privacyCost, cognitiveOverload float64) float64 {
	autonomyCost := 0.7 * math.Max(frictionAsymmetry, 0.0) * autonomyPreference
	return joined*(baselineValue+0.35*rewardIntensity) - autonomyCost - privacyCost - 0.45*cognitiveOverload
}

func main() {
	w := userWelfare(1.0, 0.45, 0.35, 0.0, 0.58, 0.05, 0.42)
	fmt.Printf("Synthetic user welfare: %.3f\n", w)
}

File: articles/behavioral-design-technology-systems/rust/interface_welfare.rs
fn user_welfare(
    joined: f64,
    baseline_value: f64,
    reward_intensity: f64,
    friction_asymmetry: f64,
    autonomy_preference: f64,
    privacy_cost: f64,
    cognitive_overload: f64,
) -> f64 {
    joined * (baseline_value + 0.35 * reward_intensity)
        - 0.7 * friction_asymmetry.max(0.0) * autonomy_preference
        - privacy_cost
        - 0.45 * cognitive_overload
}

fn main() {
    let w = user_welfare(1.0, 0.45, 0.35, 0.0, 0.58, 0.05, 0.42);
    println!("Synthetic user welfare: {:.3}", w);
}
